### PR TITLE
Make CM and Unit transforms apply more broadly

### PR DESCRIPTION
**Pull request by bernardkwok**
Update #758
- Remove restriction that a filename input with units or cm transforms on it needs to be named "file" and on a filetexture node.
- New test case is found here: resources\Materials\TestSuite\stdlib\color_management\filename_cm_test.mtlx

Source results are attached:
[Filename_CM_Test1_shaderref1_ps.glsl,txt](https://github.com/autodesk-forks/MaterialX/files/4206328/Filename_CM_Test1_shaderref1_ps.glsl.txt)
[Filename_CM_Test1_shaderref1.osl.txt](https://github.com/autodesk-forks/MaterialX/files/4206329/Filename_CM_Test1_shaderref1.osl.txt)

Render results (GLSL/OSL)
Filename_CM_Test1_shaderref1_glsl
Filename_CM_Test1_shaderref1_osl

### CODE REVIEW

File: source/MaterialXGenShader/ShaderGraph.cpp [hunk 1]
```diff
@@ -1043,43 +1044,24 @@ ShaderNode* ShaderGraph::createNode(const Node& node, GenContext& context)
     {
         for (InputPtr input : node.getInputs())
         {
-            populateInputColorTransformMap(colorManagementSystem, newNode, input, targetColorSpace);
+            ShaderInput* shaderInput = newNode->getInput(input->getName());
+            populateColorTransformMap(true, colorManagementSystem, shaderInput, input, targetColorSpace);
         }
         for (ParameterPtr parameter : node.getParameters())
         {
-            populateInputColorTransformMap(colorManagementSystem, newNode, parameter, targetColorSpace);
```
Comment (bernardkwok) [inline source/MaterialXGenShader/ShaderGraph.cpp]: Anything that is a of type "filename" is possible to transform

File: source/MaterialXGenShader/ShaderGraph.cpp [hunk 2]
```diff
@@ -1572,26 +1540,33 @@ void ShaderGraph::setVariableNames(GenContext& context)
     }
 }
 
-void ShaderGraph::populateInputColorTransformMap(ColorManagementSystemPtr colorManagementSystem, ShaderNodePtr shaderNode, ValueElementPtr input, const string& targetColorSpace)
+void ShaderGraph::populateColorTransformMap(bool asInput, ColorManagementSystemPtr colorManagementSystem, ShaderPort* shaderPort,
```
Comment (bernardkwok) [inline source/MaterialXGenShader/ShaderGraph.cpp]: This makes it so one routine can handle both input and outputs (same as what was done for units)

Comment (feldstj) [inline source/MaterialXGenShader/ShaderGraph.cpp]: nit-pick, I think the ColorManagementSystem should be the first parameter.

Comment (feldstj) [reply source/MaterialXGenShader/ShaderGraph.cpp]: asInput could probably be last, no?

**Pull request merged**